application/src/main/java/com/sanction/thunder/authentication/Key.java

```diff
@@ -5,9 +5,11 @@
 import com.google.common.base.MoreObjects;
 import com.google.common.base.Objects;
 
+import java.security.Principal;
+
 import static com.google.common.base.Preconditions.checkNotNull;
 
-public class Key {
+public class Key implements Principal {
 
   @JsonProperty("application")
```

Comment:
Not sure if this is possible but if we can get rid of all these `@JsonProperty` annotations that would be nice.